### DISCUSSION TITLE
Usage of Scrollspy within Modal doesn't work (combined from Docs examples)

### DISCUSSION THREAD

**Discussion by borysrybak** [twbs/bootstrap#37624] [Q&A] [answered]
Hi,

I'm quite new into Bootstrap, and yet I'm already so excited about its modularity and features <3 

I was experimenting recently with certain **Components** and I found out that the **Scrollspy** doesn't work properly once it's wrapped by **Modal** body (Modal itself works fine tho).

To give you a MRE - I did simply use a simple (plain) bootstrap template for HTML (css,js included), and I created something like this:

Here is a link to open/toggle modal view:
```html
<a href="#" data-bs-toggle="modal" data-bs-target="#myId"> ... </a>
```

Here is a modal part:
```html
<div class="modal fade" id="myId" tabindex="-1" aria-labelledby="exampleModalFullscreenLabel" aria-hidden="true">
      <div class="modal-dialog modal-dialog-centered">
            <div class="modal-content">
                <div class="modal-header">
                    <h1 class="modal-title fs-4" id="exampleModalFullscreenLabel">Title</h1>
                    <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
                </div>
                <div class="modal-body">
                    /* HERE COMES SCROLLSPY */
                </div>
                <div class="modal-footer">
                    <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Close</button>
                </div>
            </div>
      </div>
</div>
```
                
                
And here is a ScrollSpy part (that I taken aswell from Docs) which have been pasted within div "modal-body":
```html
<div class="row">
      <div class="col-4">
            <div id="list-example" class="list-group">
                  <a class="list-group-item list-group-item-action" href="#list-item-1">Item 1</a>
                  <a class="list-group-item list-group-item-action" href="#list-item-2">Item 2</a>
                  <a class="list-group-item list-group-item-action" href="#list-item-3">Item 3</a>
                  <a class="list-group-item list-group-item-action" href="#list-item-4">Item 4</a>
            </div>
      </div>
      <div class="col-8">
            <div data-bs-target="#list-example" data-bs-smooth-scroll="true" class="scrollspy-example" tabindex="0" style="height: 200px; overflow-y: scroll;">
                  <h4 id="list-item-1">Item 1</h4>
                  <p>Lorem ipsum ... long text...</p>
                  ...
            </div>
      </div>
</div>
```

I didn't paste rest of `<h4><p>` parts, didnt want to spam - these are take from item-1 and copy-pasted, and of course their id="list-item-X" is updated accordingly

**Observation**
What I noticed is when you paste ScrollSpy code snippet directly within `<main>`, or somewhere in so-to-speak standalone `<div class="container">` element it will act perfect as on DEMO, however pasting the example snippet within `<div class="modal-body">` (essentially Modal body) it won't work as expected: .active class is not toggled, there is no scroll animation between IDs/Headers of the H4. 

Is there any way to solve this problem? or is there any way to for example overwrite styles/variables that are messing ScrollSpy functionality while being in different component (like Modal in this is situation)?

Cheers!

**Comment by nkdas91** [comment, marked as answer]
Hi @borysrybak 

You need to call ```refresh()``` on the Scrollspy instance when the Modal is opened.

Read more here:
https://getbootstrap.com/docs/5.2/components/scrollspy/#non-visible-elements

**Reply by borysrybak** [reply]
Much appreciated! It works, thank you!
